api.ts

```diff
@@ -2203,3 +2296,49 @@ export function maybeRewriteConnectionForFormula<
   }
   return formula;
 }
+
+// This helper method finds any inline autocomplete functions in a static sync table schema.
+// These functions will need to be extracted into the "namedAutocompletes" property on the sync
+// table and replaced with strings.
+function listPropertiesWithAutocompleteFunctions(schema: ObjectSchemaDefinition<string, string>): string[] {
+  const result: string[] = [];
+  for (const propertyName of Object.keys(schema.properties)) {
+    const {autocomplete} = schema.properties[propertyName];
+    if (!autocomplete) {
+      continue;
+    }
+    if (typeof autocomplete !== 'function') {
+      continue;
+    }
+
+    result.push(propertyName);
+  }
+  return result;
+}
+
+// Finds any inline autocomplete functions within the inputSchema and replaces them
+// with strings references into the returned namedAutocompletes.
+function replaceInlineAutocompleteFunctionsWithNamedAutocompleteFunctions({
+  inputSchema, // DO NOT MUTATE inputSchema!
+  schema,
+  identityName,
+}: {
+  inputSchema: ObjectSchemaDefinition<any, any>;
+  schema: ObjectSchemaDefinition<any, any>;
+  identityName: string;
+}): SyncTableAutocompleters {
+  // Converting JS functions to strings happens on inputSchema instead of the deep copied version because the
+  // deep copy will have already thrown away any JS functions.
+  const namedAutocompletes: SyncTableAutocompleters = {};
+
+  for (const propertyName of listPropertiesWithAutocompleteFunctions(inputSchema)) {
+    schema.properties[propertyName].autocomplete = propertyName as AutocompleteReference;
+    namedAutocompletes[propertyName] = makePropertyAutocompleteFormula({
+      execute: inputSchema.properties[propertyName].autocomplete as any,
```

Comment:
Seems like in a future PR we can support static declarations of enumerable values by just taking an array that's given as the `autocomplete` value and wrapping it in a function that returns it here. Which is what I think we do for parameter autocomplete already.

Reply:
Technically this PR already supports static declarations of enumerable values, although I haven't added that support into the `coda` code.

I haven't wrapped static value lists in a formula because it seems like we might want to be able to just read them directly in the doc without invoking a pack over the network

The `T[]` below is what allows for static declarations of values:
```
/**
 * A function or set of values to return for property autocomplete.
 * @hidden
 */
export type PropertySchemaAutocomplete<T extends PackFormulaResult> =
  | PropertyAutocompleteMetadataFunction<T[]>
  | T[]
  | AutocompleteType
  | AutocompleteReference;
```